modules/tensor_mechanics/src/materials/ADComputeFiniteStrainElasticStress.C

```diff
@@ -26,6 +26,8 @@ ADComputeFiniteStrainElasticStress::ADComputeFiniteStrainElasticStress(
     _elasticity_tensor_name(_base_name + "elasticity_tensor"),
     _elasticity_tensor(getADMaterialProperty<RankFourTensor>(_elasticity_tensor_name)),
     _strain_increment(getADMaterialPropertyByName<RankTwoTensor>(_base_name + "strain_increment")),
+    _rotation_total(declareADProperty<RankTwoTensor>("rotation_total")),
```

Comment:
Is there any way to store a pointer rather than a reference, and only get the pointer for the old rotation if it's not guaranteed isotropic?  That would save allocating another RankTwoTensor for every material point, but I'm not sure whether we can do that.

Reply:
If I understand well. Conditionally defining pointers would be possible if knew upon object construction whether the tensor is guaranteed to be isotropic, but, at that point, that information isn't available.

Reply:
OK, probably not much we can do about this then.